dashboard/modules/reporter/reporter_head.py

```diff
@@ -411,6 +411,103 @@ async def get_task_cpu_profile(self, req) -> aiohttp.web.Response:
             headers={"Content-Type": "text/html"},
         )
 
+    @routes.get("/task/memory_profile")
+    async def task_memory_profile(self, req) -> aiohttp.web.Response:
+        """
+        Retrieves the memory profile for a specific task.
+        Note that one worker process works on one task at a time
+        or one worker works on multiple async tasks.
+
+        Args:
+            req (aiohttp.web.Request): The HTTP request object.
+
+        Returns:
+            aiohttp.web.Response: The HTTP response containing the memory profile data.
+
+        Raises:
+            ValueError: If the "task_id" parameter is
+            missing in the request query.
+            ValueError: If the "attempt_number" parameter is
+            missing in the request query.
+            ValueError: If the maximum duration allowed is exceeded.
+            ValueError: If the worker begins working on
+            another task during the profile retrieval.
+            aiohttp.web.HTTPInternalServerError: If there is
+            an internal server error during the profile retrieval.
+        """
+        if "task_id" not in req.query:
+            raise ValueError("task_id is required")
+        if "attempt_number" not in req.query:
+            raise ValueError("task's attempt number is required")
+        if "node_id" not in req.query:
+            raise ValueError("node_id is required")
+
+        task_id = req.query.get("task_id")
+        attempt_number = req.query.get("attempt_number")
+        node_id = req.query.get("node_id")
+
+        duration = int(req.query.get("duration", 30))
+
+        # Default not using `--native`, `--leaks` and `--format` for profiling
+        format = req.query.get("format", "flamegraph")
+        native = req.query.get("native", False) == "1"
+        leaks = req.query.get("leaks", False) == "1"
+
+        ip = DataSource.node_id_to_ip[node_id]
+        reporter_stub = self._stubs[ip]
+
+        try:
+            (pid, _) = await self.get_worker_details_for_running_task(
+                task_id, attempt_number
+            )
+        except ValueError as e:
+            raise aiohttp.web.HTTPInternalServerError(text=str(e))
+
+        logger.info(
+            "Retrieving memory profiling request to {}:{} for {}".format(
+                ip, pid, task_id
+            )
+        )
+
+        reply = await reporter_stub.MemoryProfiling(
+            reporter_pb2.MemoryProfilingRequest(
+                pid=pid, format=format, leaks=leaks, duration=duration, native=native
+            )
+        )
+
+        """
+            In order to truly confirm whether there are any other tasks
```

Comment:
Out of curiosity, why does it matter whether there are any other tasks running during profiling?

Reply:
I'm not too sure about this rn, just followed the previous implementation for task profiler by @chaowanggg 
I can try investigating this and get back to you